app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -36,14 +37,14 @@ fun Nav() {
                 RegisterScreen(
                     state = state,
                     onEvent = registerViewModel::onEvent,
-                    navController = navController
+                    context = context
                 )
             }
         }
     }
 }
 @Composable
-inline fun <reified T : ViewModel> NavBackStackEntry.loginViewModel(
+inline fun <reified T : ViewModel> NavBackStackEntry.sharedViewModel(
     navController: NavController,
 ): T {
     val navGraphRoute = destination.parent?.route ?: return hiltViewModel()
```

Comment:
A shared VM only makes sense if you need to share state between multiple screens. However, for authentication that's not really needed, since each screen has its own state. Sharing the state will technically work, but it mixes responsibilities together in one ViewModel. 
Instead, I'd recommend to have a separate `LoginViewModel` and `RegisterViewModel` which are only bound to their corresponding screen.